apps/opik-frontend/src/components/pages/PlaygroundPage/PlaygroundOutputs/PlaygroundOutputScores/PlaygroundOutputScores.tsx

```diff
@@ -0,0 +1,356 @@
+  // Note: We deduplicate by score name because TraceFeedbackScore doesn't include rule_id,
+  // so if multiple rules produce the same metric name, we can only show one entry
+  const expectedMetrics = useMemo(() => {
+    const metrics: Array<{
+      scoreName: string;
+      score: TraceFeedbackScore | null;
+    }> = [];
+    const seenScoreNames = new Set<string>();
+
+    // If selectedRuleIds is null/undefined (meaning "all" was selected or legacy data),
+    // only show scores that actually exist in the trace - no loading spinners
+    if (selectingAllRules) {
+      for (const score of feedbackScores) {
+        if (!seenScoreNames.has(score.name)) {
+          seenScoreNames.add(score.name);
+          metrics.push({
+            scoreName: score.name,
+            score,
+          });
+        }
+      }
+    } else {
+      // Specific rules were selected - need rules to load to know expected scores
+      if (!rulesLoaded) return [];
+
+      for (const rule of selectedRules) {
+        const scoreNames = getScoreNamesFromRule(rule);
+        for (const scoreName of scoreNames) {
+          // Deduplicate: if multiple rules produce the same metric name, show only once
+          if (!seenScoreNames.has(scoreName)) {
+            seenScoreNames.add(scoreName);
+            metrics.push({
+              scoreName,
+              score: scoresByName[scoreName] || null,
+            });
+          }
+        }
+      }
+    }
+
+    // Sort by score name alphabetically
+    return metrics.sort((a, b) => a.scoreName.localeCompare(b.scoreName));
+  }, [
+    rulesLoaded,
+    selectedRules,
+    scoresByName,
+    selectingAllRules,
+    feedbackScores,
+  ]);
+
+  // Don't render anything if no rules are selected
+  if (!hasSelectedRules) {
+    return null;
+  }
+
+  // For specific rules selected: show when output is ready (even before traceId)
+  // For "all" selection: only show when we have traceId (to get actual scores)
+  const hasSpecificRules = !selectingAllRules;
+  const canShowMetrics = traceId || (outputReady && hasSpecificRules);
+
+  if (!canShowMetrics) {
+    return null;
+  }
+
+  // Don't render if rules loaded but no expected metrics (and we have specific rules selected)
+  if (
+    rulesLoaded &&
+    expectedMetrics.length === 0 &&
+    selectedRuleIds !== null &&
+    selectedRuleIds !== undefined
+  ) {
+    return null;
+  }
+
+  const visibleMetrics = expectedMetrics.slice(0, MAX_VISIBLE_METRICS);
+  const hiddenMetrics = expectedMetrics.slice(MAX_VISIBLE_METRICS);
+  const remainingCount = hiddenMetrics.length;
+
+  const renderMetric = ({
+    scoreName,
+    score,
+  }: {
+    scoreName: string;
+    score: TraceFeedbackScore | null;
+  }) => {
+    const variant = generateTagVariant(scoreName);
+    const color = (variant && TAG_VARIANTS_COLOR_MAP[variant]) || "#64748b";
+
+    // Score has loaded - show the actual value
+    if (score) {
+      return (
+        <FeedbackScoreTag
```

Comment:
Tested on latest testing-environment:

Don't see the loading state of the MetricTags

Just waiting few seconds, and then they appear:

https://github.com/user-attachments/assets/7f3b304b-2b4c-45c0-8d74-5eeb7c825b05